src/client.rs

```diff
@@ -0,0 +1,67 @@
+use crate::prelude::*;
+use jsonrpsee::ws_client::WsClientBuilder;
+use subxt::backend::rpc::RpcClient as RawRpcClient;
+
+/// Wraps the subxt interfaces to make it easy to use for the staking-miner.
+#[derive(Clone, Debug)]
```

Comment:
subxt's APIs has been modified a bit, thus this type is introduced to provide a wrapper for the APIs that the staking miner needs.